### ISSUE TITLE
manual_non_exhaustive should not warn on a used enum variant

### ISSUE THREAD

**Issue by DomWilliams0** [rust-lang/rust-clippy#5714]
I have an enum similar to the following:

```rust

enum State {
        /// Temporary value to use in place of uninitialized memory, for safe handling of panics
        #[doc(hidden)]
        _Ephemeral,

        Uninit(...),
        Init(...),
    }
```

Clippy suggests that I remove the `_Ephemeral` variant in place of `#[non_exhaustive]`, which wouldn't work in this case as `_Ephemeral` is used, so removing it would be an error. I think the lint should only complain if the variant isn't actually used?

```
cargo clippy -V
clippy 0.0.212 (feb3536 2020-06-09)
```
Thanks!

**Comment by flip1995** [comment]
The problem is that we can't detect if this value is used, since this would require a global analysis. I'm interested if more people have this problem or if it is a special case. If more people have this problem, the only viable solution would be to move this lint to `pedantic`.

**Comment by ebroto** [comment]
Another possibility would be to split the lint. I think that for structs it could still work well as it requires the type of the private member to be the unit type, and is safer to assume that the "manual non-exhaustive" pattern was being used.
